repos/system_upgrade/common/actors/checksaphana/libraries/checksaphana.py

```diff
@@ -195,33 +199,39 @@ def version2_check(info):
 
 
 def platform_check():
-    """ Creates an inhibitor report in case the system is not running on x86_64 """
-    if not architecture.matches_architecture(architecture.ARCH_X86_64):
-        if version.get_target_major_version() == '8':
-            elink = reporting.ExternalLink(
-                url='https://access.redhat.com/solutions/5533441',
-                title='How do I upgrade from Red Hat Enterprise Linux 7 to Red Hat Enterprise Linux 8 with SAP HANA')
-        else:
-            elink = reporting.ExternalLink(
-                url='https://access.redhat.com/solutions/6980855',
-                title='How to in-place upgrade SAP environments from RHEL 8 to RHEL 9')
-
-        reporting.create_report([
-            reporting.Title('SAP HANA upgrades are only supported on X86_64 systems'),
-            reporting.Summary(
-                ('Upgrades for SAP HANA are only supported on X86_64 systems.'
-                 ' For more information please consult the documentation.')
-            ),
-            reporting.Severity(reporting.Severity.HIGH),
-            reporting.Groups([reporting.Groups.SANITY]),
-            reporting.Groups([reporting.Groups.INHIBITOR]),
-            reporting.Audience('sysadmin'),
-            elink,
-        ])
-        return False
-
-    return True
+    """ Creates an inhibitor report in case the SAP HANA system is not running
+        - on x86_64 when upgrding from RHEL7 to RHEL8
+        - on x86_64 or ppc64le when upgrading from RHEL8 to RHEL9 """
+
+    if version.get_target_major_version() == '8' and \
+       api.current_actor().configuration.architecture not in SAP_HANA_7TO8_ARCH:
+        elink = reporting.ExternalLink(
+            url='https://access.redhat.com/solutions/5154031',
+            title='How to in-place upgrade SAP environments from RHEL 7 to RHEL 8')
+        supported_arch = SAP_HANA_7TO8_ARCH
+    elif version.get_target_major_version() == '9' and \
+       api.current_actor().configuration.architecture not in SAP_HANA_8TO9_ARCH:
+        elink = reporting.ExternalLink(
+            url='https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_for_sap_solutions/9/html/how_to_in-place_upgrade_sap_environments_from_rhel_8_to_rhel_9/index',  # noqa: E501; pylint: disable=line-too-long
+            title='How to in-place upgrade SAP environments from RHEL 8 to RHEL 9')
+        supported_arch = SAP_HANA_8TO9_ARCH
+    else:
+        return True
 
+    reporting.create_report([
+        reporting.Title('Unsupported architecture for SAP HANA systems on target version'),
+        reporting.Summary(
+            (f'For RHEL{version.get_target_major_version()}.* target version \
```

Comment:
do not use format strings that does not exist on python where the code is running